pkg/lib/queueinformer/jitter.go

```diff
@@ -1,32 +1,42 @@
 package queueinformer
 
 import (
+	"math"
 	"math/rand"
 	"time"
 )
 
 const DefaultResyncPeriod = 15 * time.Minute
 
+type float64er interface {
+	Float64() float64
+}
+
+type realFloat64er struct{}
+
+func (realFloat64er) Float64() float64 {
+	return rand.Float64()
+}
+
 // ResyncWithJitter takes a resync interval and adds jitter within a percent difference.
 // factor is a value between 0 and 1 indicating the amount of jitter
 // a factor of 0.2 and a period of 10m will have a range of 8 to 12 minutes (20%)
 func ResyncWithJitter(resyncPeriod time.Duration, factor float64) func() time.Duration {
+	return resyncWithJitter(resyncPeriod, factor, realFloat64er{})
+}
+
+func resyncWithJitter(period time.Duration, factor float64, rand float64er) func() time.Duration {
 	return func() time.Duration {
-		if factor < 0.0 || factor > 1.0 {
-			return resyncPeriod
-		}
-		if resyncPeriod < 0.0 {
+		if period < 0.0 {
```

Comment:
Should this maybe be `period <= 0.0`? This would help with the extreme case where the duration comes out to zero that we have been seeing.

Reply:
It might be better to set a higher floor on this because 1ns is effectively as bad as 0. I left it alone for now.